### PR TITLE
Bump bootstrap compiler to beta 1.53.0

### PR DESCRIPTION
**Pull request by emilyalbini**
This PR bumps the bootstrap compiler to version 1.53.0 beta, as part of our usual release process (this was supposed to be Wednesday's step, but creating the beta release took longer than expected).

The PR also includes the "Bootstrap: skip rustdoc fingerprint for building docs" commit, see the reasoning [on Zulip](https://zulip-archive.rust-lang.org/241545trelease/88450153betabootstrap.html).

r? @Mark-Simulacrum

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
r=me modulo CI

File: compiler/rustc_apfloat/src/lib.rs [hunk 1]
```diff
@@ -35,7 +35,6 @@
 #![forbid(unsafe_code)]
 #![feature(iter_zip)]
 #![feature(nll)]
-#![cfg_attr(bootstrap, feature(or_patterns))]
```
Comment (eddyb) [inline compiler/rustc_apfloat/src/lib.rs:38]: Heh so that's how I got pinged. At least the ping works. (And this change is obviously fine)

**Comment by pietroalbini** [comment]
Rebased to fix the merge conflict.

@bors r=Mark-Simulacrum

**Comment by pietroalbini** [comment]
@bors r-

I'm a bit puzzled by this failure, `#![feature(no_coverage)]` is present in `library/core/src/lib.rs`...

**Comment by jyn514** [comment]
@pietroalbini https://github.com/rust-lang/rust/pull/84871 may need to be backported

**Comment by Mark-Simulacrum** [comment]
https://github.com/rust-lang/rust/pull/85588 includes #84871 backport, will rebase this once that lands.

**Comment by Mark-Simulacrum** [comment]
Rebased along with a bump to the new beta (which included the relevant backport for the coverage attribute).

**Pull request merged**